backend/inspirehep/records/api/mixins.py

```diff
@@ -364,17 +370,25 @@ def get_all_connected_records_uuids_of_modified_authors(self):
                 )
                 .distinct(RecordsAuthors.record_id)
                 .all()
-            ]
-        return differed_papers_uuids
+            }
+            connected_papers = citers | self_cited
+
+        return connected_papers
 
     def get_all_connected_records_uuids_of_modified_collaborations(self):
         prev_version = self._previous_version
         current_collaborations = set(self.get_collaborations_values())
         old_collaborations = set(prev_version.get_collaborations_values())
-        diff = list(current_collaborations.symmetric_difference(old_collaborations))
-        differed_papers_uuids = []
+        diff = current_collaborations.symmetric_difference(old_collaborations)
+        connected_papers = set()
         if diff:
-            differed_papers_uuids = [
+            citers = {
+                citer[0]
+                for citer in RecordCitations.query.filter_by(cited_id=self.id)
+                .with_entities("citer_id")
+                .all()
+            }
```

Comment:
you're doing the same query twice potentially, also for modified authors (not dramatic, but suboptimal).

Reply:
This query is small and fast and it's easy to do it this way, at the end there won't be any duplicates as results are de-duplicated